src/main/scala/com/gu/invoicing/nextinvoicedate/Lambda.scala

```diff
@@ -0,0 +1,30 @@
+package com.gu.invoicing.nextinvoicedate
+
+import java.io.{InputStream, OutputStream}
+import com.gu.invoicing.nextinvoicedate.Model._
+import com.gu.invoicing.nextinvoicedate.Impl._
+import com.gu.invoicing.nextinvoicedate.Program._
+import scala.util.chaining._
+import com.gu.spy._
+
+/**
+ * Example test event for running the lambda from AWS Console
+ * {
+ *   "pathParameters": {
+ *     "subscriptionNumber": "A-S00000000"
+ *   }
+ * }
+ */
+object Lambda {
+  def handleRequest(input: InputStream, output: OutputStream): Unit =
+    input
+      .pipe { read[ApiGatewayInput](_) }
+      .pipe { NextInvoiceDateInput.apply }
+      .tap  { info[NextInvoiceDateInput] }
+      .pipe { program }
+      .tap  { info[NextInvoiceDateOutput] }
+      .pipe { invoiceDateOutput => ApiGatewayOutput(200, write(invoiceDateOutput)) }
+      .pipe { write(_) }
+      .pipe { _.getBytes }
+      .pipe { output.write }
+}
```

Comment:
You can use API gateway event instances as input and output of a handler, to make it a bit easier

Reply:
I am trying not to depend on external dependencies to keep the package small which helps with warmup time. This is a one-off pain so